servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1414,9 +1422,22 @@ private void setStateInternal(long fileId, long opTimeMs, SetStateOptions option
         mTTLBuckets.insert(file);
       }
     }
+    if (options.hasPersisted()) {
+      Preconditions.checkArgument(inode.isFile(), PreconditionMessage.PERSIST_ONLY_FOR_FILE);
+      Preconditions.checkArgument(((InodeFile) inode).isCompleted(),
+          PreconditionMessage.FILE_TO_PERSIST_MUST_BE_COMPLETE);
+      InodeFile file = (InodeFile) inode;
+      if (!file.isPersisted()) {
+        file.setPersisted(true);
```

Comment:
This code assumes that the input persisted option flag is always true? Could that be confusing?